worker/codalabworker/local_run/local_run_manager.py

```diff
@@ -360,3 +360,26 @@ def memory_bytes(self):
         except ValueError:
             # Fallback to sysctl when os.sysconf('SC_PHYS_PAGES') fails on OS X
             return int(check_output(['sysctl', '-n', 'hw.memsize']).strip())
+
+    @property
+    def free_disk_bytes(self):
+        """
+        Available disk space by bytes of this RunManager.
+        """
+        try:
+            p = Popen(["df", self._work_dir], stdout=PIPE)
+            output, error = p.communicate()
+            # Return -1 when there is an error.
+            if error:
+                logger.info(error.strip())
+                return -1
+
+            if output:
+                lines = output.decode().split("\n")
+                index = lines[0].split().index("Available")
+                # We convert the original result from df command in unit of 1Kbyte blocks into bytes.
+                return int(lines[1].split()[index]) * 1024
+
+        except OSError as e:
```

Comment:
This isn't the only error that can happen...  if the format of `df` changes for some reason, then the parsing logic can fail too

Reply:
I also thought about the problems of parsing `df` result. `shutil` in Python3 provide `disk_usage` function which will return same result as `df`, but not in Python2. `psutil` in Python2 provides the same function. Not sure if this would be a better option?